.github/workflows/static-tests.yml

```diff
@@ -0,0 +1,36 @@
+name: "static tests"
+on: [pull_request, push]
+env:
+  TRAVIS_BRANCH: "master"
+jobs:
+  statics-tests:
+    name: Static tests
+    runs-on: ubuntu-latest
+    env:
+      PYTHON_VERSION: 3.6
+      AIRFLOW_MOUNT_SOURCE_DIR_FOR_STATIC_CHECKS: "true"
+      TRAVIS_JOB_NAME: "Static"
+    strategy:
+      matrix:
+        scripts: [ci_run_static_checks_pylint_tests.sh,
+                  ci_mypy.sh,
+                  ci_fix_ownership.sh,
```

Comment:
Been there done that @dimberman .

I think this will not increase the speed. It might even slow it down. 

There is quite an overhead introduced by creating a new runner (it works similarly in Travis and GA last time I checked). Vast majority of the tests - including mypy, llake8, bat_tests are fast and it makes no sense to spin separate worker just to run the single static check.. The ones that take vast (like 90%) of time are pylint_main and pylint_test), I experimented with it before and the best overall execution time was what was there before:

1, job - run all tests but pylint_tests
2, Run pylint_tests

Further fragmentation introduces much more overhead for worker creation and elapsed time to execute the builds was much longer. With this approach you can gain at most single seconds in the best case. 

What would ultimately help if is we finish all pylint tests and introduce paralllelism in pylint tests (the tests can run muich faster but currently we have uncertainty of cycles introduced because some of our code is excluded from pylint and contains cycles, Those cycles are randomly appearing when people make their PRs so Pylint tests are now set to run without parallelism.

Reply:
Also one more comment  why this will be much, much, much slower - some of the tests (mypy, flake, pylint) require CI image to run (to keep consistent environment). If you split it to separate jobs, each of them will pull the image before running and will rebuild it with latest sources (to make sure they are using latest setup.py for tests). When you run mypy + flake + pylint in a single pre-commit run, they will all download the image and potentially build the image with the latest setup.py - this adds even more "per-job" overhead.

Reply:
I think what can help is to split static checks into action steps (should be easy with pre-commit). This will probably decrease performance but it will increase the readability of the output (you will see which step exactly failed).

Reply:
I think what can help better is to keep most of the checks together but to get rid of the --verbose flag. The flag was introduced to see time of execution of each step. But... it's not needed any more in GA - in GA you can click the checkbox and see the timestamp for each line of the log. Once you disable the --verbose flag, it is much more obvious which step is failing because you will see a wall of green 'Passed" broken by some "Failed"  (the output of each command is hidden if the result is "Passed"). This will be much better approach.

Reply:
At least we can move pylint and mypy to separate step. In this way the CI will fail faster if there are any other checks failing. In this way the feedback loop will be definitely faster.